Lucid/Core/RelationshipController.swift

```diff
@@ -124,6 +132,18 @@ public final class ThreadSafeGraph<Graph>: MutableGraph where Graph: MutableGrap
         return dispatchQueue.sync { _value.contains(identifier) }
     }
 
+    public func setEndpointResultMetadata(_ metadata: EndpointResultMetadata) {
+        dispatchQueue.async {
+            self._value.setEndpointResultMetadata(metadata)
+        }
+    }
+
+    public func metadata<E>() -> Metadata<E>? where E : Entity {
+        dispatchQueue.sync {
+            return self._value.metadata()
+        }
```

Comment:
Optional, this can also be written as:

```swift
dispatchQueue.sync { _value.metadata() }
```

The `self.` isn't required in `sync` blocks as there is no retention on the parent object.